model/map/src/main/java/org/keycloak/models/map/realm/MapRealmAdapter.java

```diff
@@ -0,0 +1,1547 @@
+        entity.setAccessTokenLifespan(seconds);
+    }
+
+    @Override
+    public int getAccessTokenLifespanForImplicitFlow() {
+        return entity.getAccessTokenLifespanForImplicitFlow();
+    }
+
+    @Override
+    public void setAccessTokenLifespanForImplicitFlow(int seconds) {
+        entity.setAccessTokenLifespanForImplicitFlow(seconds);
+    }
+
+    @Override
+    public int getAccessCodeLifespan() {
+        return entity.getAccessCodeLifespan();
+    }
+
+    @Override
+    public void setAccessCodeLifespan(int seconds) {
+        entity.setAccessCodeLifespan(seconds);
+    }
+
+    @Override
+    public int getAccessCodeLifespanUserAction() {
+        return entity.getAccessCodeLifespanUserAction();
+    }
+
+    @Override
+    public void setAccessCodeLifespanUserAction(int seconds) {
+        entity.setAccessCodeLifespanUserAction(seconds);
+    }
+
+    @Override
+    public int getAccessCodeLifespanLogin() {
+        return entity.getAccessCodeLifespanLogin();
+    }
+
+    @Override
+    public void setAccessCodeLifespanLogin(int seconds) {
+        entity.setAccessCodeLifespanLogin(seconds);
+    }
+
+    @Override
+    public int getActionTokenGeneratedByAdminLifespan() {
+        return entity.getActionTokenGeneratedByAdminLifespan();
+    }
+
+    @Override
+    public void setActionTokenGeneratedByAdminLifespan(int seconds) {
+        entity.setActionTokenGeneratedByAdminLifespan(seconds);
+    }
+
+    @Override
+    public int getActionTokenGeneratedByUserLifespan() {
+        return entity.getActionTokenGeneratedByUserLifespan();
+    }
+
+    @Override
+    public void setActionTokenGeneratedByUserLifespan(int seconds) {
+        entity.setActionTokenGeneratedByUserLifespan(seconds);
+    }
+
+    @Override
+    public int getActionTokenGeneratedByUserLifespan(String actionTokenType) {
+        return getAttribute(ACTION_TOKEN_GENERATED_BY_USER_LIFESPAN + actionTokenType, getAccessCodeLifespanUserAction());
+    }
+
+    @Override
+    public void setActionTokenGeneratedByUserLifespan(String actionTokenType, Integer seconds) {
+        if (actionTokenType != null && ! actionTokenType.isEmpty() && seconds != null) {
+            setAttribute(ACTION_TOKEN_GENERATED_BY_USER_LIFESPAN + actionTokenType, seconds);
+        }
+    }
+
+    @Override
+    public Map<String, Integer> getUserActionTokenLifespans() {
+        Map<String, Integer> tokenLifespans = entity.getAttributes().entrySet().stream()
+                .filter(Objects::nonNull)
+                .filter(entry -> nonNull(entry.getValue()))
+                .filter(entry -> entry.getKey().startsWith(ACTION_TOKEN_GENERATED_BY_USER_LIFESPAN))
+                .collect(Collectors.toMap(
+                        entry -> entry.getKey().replace(ACTION_TOKEN_GENERATED_BY_USER_LIFESPAN, ""),
+                        entry -> Integer.valueOf(entry.getValue())));
+
+        tokenLifespans.put("", getActionTokenGeneratedByUserLifespan());
```

Comment:
Why this line needs to be here? I don't see it the JPA implementation and particularly I don't understand the empty string as a key.

Reply:
There are `RealmModel.setActionTokenGeneratedByUserLifespan(int)` and `RealmModel.getActionTokenGeneratedByUserLifespan()` methods for setting and getting "default" actionTokenGeneratedByUserLifespan. The "default" value is JPA is stored in attributes with key `RealmAttributes.ACTION_TOKEN_GENERATED_BY_USER_LIFESPAN`. Here it's stored as: https://github.com/vramik/keycloak/blob/KEYCLOAK-14552-realm-map-store/model/map/src/main/java/org/keycloak/models/map/realm/AbstractRealmEntity.java#L64

The `getUserActionTokenLifespans()` should return map where key is the actionTokenId string. The "default" value is represented by empty string (it's the same as JPA). That's why it needs to be called here like this. 

But when I wrote this I had also other things stored directly as variables not in attributes, then I has to change it back. 

It might be needed to edit this one as well, but I'm not sure tbh.